Adds `get_contours` and supports "method" key word a keyword argument to select between `contourpy` and `scikit-image` (#7760)
sunpy/sunpy: changelog/7760.breaking.rst

> The `skimage` library has been replaced with `ContourPy` and creating contours

Reviewer: ```suggestion
Replaced the use of scikit-image with ContourPy within `sunpy.map.Map.contour`.
There is no change to returned contours but offers a faster way to contour. 
```

> `sunpy.map.GenericMap.contour` now internally uses ContourPy instead of scikit-image for improved performance.

Reviewer: Can the text in this changelog be adjusted for Cadair's comment above?

Author: Is it fine now?

> Replaced the use of scikit-image with ContourPy within `sunpy.map.Map.contour`.
> There is no change to returned contours but offers a faster way to contour.

Reviewer 2: ```suggestion
`sunpy.map.GenericMap.contour` now internally uses ContourPy instead of scikit-image for improved performance.
The API for the method has not changed, but as before, additional keyword arguments are passed through to the underlying function, so some such keyword arguments may no longer be supported.
```

Reviewer 2: This also fixes the broken ref

> :meth:`sunpy.map.GenericMap.contour` now uses ContourPy instead of scikit-image for improved performance and consistency with :meth:`sunpy.map.GenericMap.draw_contours`.

Reviewer 2: The changelog entry needs to warn that the returned contours will likely be different (in order, if not in values)

> `sunpy.map.GenericMap.contour` now uses ContourPy instead of scikit-image for faster performance and consistency with `sunpy.map.GenericMap.draw_contour`, reducing reliance on scikit-image.

Reviewer: ```suggestion
`sunpy.map.GenericMap.contour` now uses ContourPy instead of scikit-image for faster performance and consistency with `sunpy.map.GenericMap.draw_contour`.
```
Is it actually faster? We can't say this if we don't verify it.

Author: Alright, how about 'improved' instead of 'faster' as suggested earlier

Reviewer: Improved how?

Author: I don't know, I tried looking for evidence but didn't find any, so what should I update changelog to?

Reviewer: Either you drop the mention of performance or you benchmark it. That is our two choices.

Author: **Performance Test**
```
import cProfile

test_code = '''

import matplotlib.pyplot as plt

import astropy.units as u
for x in range(25):
    import sunpy.map
    from sunpy.data.sample import AIA_193_IMAGE
    aiamap = sunpy.map.Map(AIA_193_IMAGE)
    contours = aiamap.contour(50000 * u.DN)
'''

if __name__ == "__main__":
    cProfile.run(test_code)
```
**Using Scikit-Image**
`8894305 function calls (18373215 primitive calls) in 62.470 seconds`
**Using ContourPy**
`17448686 function calls (16975009 primitive calls) in 58.905 seconds`

Reviewer: That is better. Improved it is.

> `sunpy.map.GenericMap.contour` now uses ContourPy instead of scikit-image for faster performance and consistency with `sunpy.map.GenericMap.draw_contour`, reducing reliance on scikit-image.

Reviewer: <unknown>:7: WARNING: py:obj reference target not found: sunpy.map.GenericMap.draw_contour [ref.obj]

Author: Its fixed now

:meth:`sunpy.map.GenericMap.contour` now uses ContourPy instead of scikit-image for improved performance and consistency with :meth:`sunpy.map.GenericMap.draw_contours`.

> This does mean that both that the returned contours may differ in order and values compared to the scikit-image version and as that additional keyword arguments are passed through to the underlying function, specific keyword arguments may no longer be supported with ContourPy.

Reviewer 3: What does this mean? Is this saying that the result of the contour method won't be the same after switching to contourpy? If so, isn't this a breaking change?

> :meth:`sunpy.map.GenericMap.contour` now uses ContourPy instead of scikit-image for improved performance and consistency with :meth:`sunpy.map.GenericMap.draw_contours`.
> The API for the method has not changed, but as before, additional keyword arguments are passed through to the underlying function, so some such keyword arguments may no longer be supported.
> The returned contours may also differ in order and values compared to the previous implementation.

Reviewer: ```suggestion
:meth:`sunpy.map.GenericMap.contour` now uses ContourPy instead of scikit-image for improved performance and consistency with :meth:`sunpy.map.GenericMap.draw_contours`.

This does mean that both that the returned contours may differ in order and values compared to the scikit-image version and as that additional keyword arguments are passed through to the underlying function, specific keyword arguments may no longer be supported with ContourPy.
```
